x/btcstaking/keeper/hooks.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"context"
+
+	ltypes "github.com/babylonlabs-io/babylon/x/btclightclient/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"go.uber.org/zap"
+)
+
+// HandledHooks Helper interface to ensure Hooks implements
+// the btclightclient hook
+type HandledHooks interface {
+	ltypes.BTCLightClientHooks
+}
+
+type Hooks struct {
+	k Keeper
+}
+
+var _ HandledHooks = Hooks{}
+
+func (k Keeper) Hooks() Hooks { return Hooks{k} }
+
+// AfterBTCRollBack updates the Largest BTC reorg if it is higher than the current one
+// to later verify in EndBlocker.
+func (h Hooks) AfterBTCRollBack(goCtx context.Context, rollbackFrom, rollbackTo *ltypes.BTCHeaderInfo) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// sanity checks for rollback values
+	if rollbackFrom == nil {
```

Comment:
hmm I feel like all those checks i.e
- `if rollbackFrom == nil`
- `if rollbackTo == nil`
- `rollbackTo.Height >= rollbackFrom.Height`

Are kind of masking potential bugs in btclightclient. 

Wdyt about:
- documenting those invariants on light client side i.e in `func (k Keeper) AfterBTCRollBack(ctx context.Context, rollbackFrom, rollbackTo *types.BTCHeaderInfo) ` 
- here either assuming those holds or having separate function `checkInvariants` which panics if those are not fulfilled.